src/MongoDB.Driver.Core/ChangeStreamDocumentSerializer.cs

```diff
@@ -50,6 +50,14 @@ public class ChangeStreamDocumentSerializer<TDocument> : BsonDocumentBackedClass
             RegisterMember("UpdateDescription", "updateDescription", ChangeStreamUpdateDescriptionSerializer.Instance);
         }
 
+        // public methods
+        /// <inheritdoc />
+        public override ChangeStreamDocument<TDocument> Deserialize(BsonDeserializationContext context, BsonDeserializationArgs args)
+        {
+            context = context.With(b => b.AllowDuplicateElementNames = true);
+            return base.Deserialize(context, args);
```

Comment:
This change allows a change stream message from the server containing duplicate element names to be deserialized without throwing an exception.